257 classic assert greater and less (#260)
nunit/nunit.analyzers: documentation/NUnit2038.md

# NUnit2038
## Consider using Assert.That(actual, Is.InstanceOf(expected)) instead of Assert.IsInstanceOf(expected, actual).

| Topic    | Value
| :--      | :--
| Id       | NUnit2038
| Severity | Info
| Enabled  | True
| Category | Assertion
| Code     | [ClassicModelAssertUsageAnalyzer](https://github.com/nunit/nunit.analyzers/blob/master/src/nunit.analyzers/ClassicModelAssertUsage/ClassicModelAssertUsageAnalyzer.cs)

## Description

Consider using the constraint model, Assert.That(actual, Is.InstanceOf(expected)), instead of the classic model, Assert.IsInstanceOf(expected, actual).

## Motivation

> The assert `Assert.InstanceOf` from the classic Assert model makes it easy to confuse the `expected` and the `actual` argument,
> so this analyzer marks usages of `Assert.InstanceOf`.

Reviewer: ```suggestion
The assert `Assert.IsInstanceOf` from the classic Assert model makes it easy to confuse the `expected` and the `actual` argument,
so this analyzer marks usages of `Assert.IsInstanceOf`.
```

```csharp
[Test]
public void Test()
{

>     Assert.InstanceOf(expected, actual);

Reviewer: ```suggestion
    Assert.IsInstanceOf(expected, actual);
```

}
```

## How to fix violations

> The analyzer comes with a code fix that will replace `Assert.AreSame(expected, actual)` with
> `Assert.That(actual, Is.SameAs(expected))`. So the code block above will be changed into.

Reviewer: ```suggestion
The analyzer comes with a code fix that will replace `Assert.IsInstanceOf(expected, actual)` with
`Assert.That(actual, Is.InstanceOf(expected))`. So the code block above will be changed into.
```
